recipes/openvino/all/conanfile.py

```diff
@@ -0,0 +1,388 @@
+        toolchain.cache_variables["ENABLE_SYSTEM_PUGIXML"] = True
+        if self._protobuf_required:
+            toolchain.cache_variables["ENABLE_SYSTEM_PROTOBUF"] = True
+        if self.options.enable_tf_frontend:
+            toolchain.cache_variables["ENABLE_SYSTEM_SNAPPY"] = True
+        if self.options.enable_tf_lite_frontend:
+            toolchain.cache_variables["ENABLE_SYSTEM_FLATBUFFERS"] = True
+        # misc
+        if self._preprocessing_available:
+            toolchain.cache_variables["ENABLE_GAPI_PREPROCESSING"] = True
+        toolchain.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
+        toolchain.cache_variables["CPACK_GENERATOR"] = "CONAN"
+        toolchain.cache_variables["ENABLE_PROFILING_ITT"] = False
+        toolchain.cache_variables["ENABLE_PYTHON"] = False
+        toolchain.cache_variables["ENABLE_PROXY"] = False
+        toolchain.cache_variables["ENABLE_WHEEL"] = False
+        toolchain.cache_variables["ENABLE_CPPLINT"] = False
+        toolchain.cache_variables["ENABLE_NCC_STYLE"] = False
+        toolchain.cache_variables["ENABLE_SAMPLES"] = False
+        toolchain.cache_variables["ENABLE_TEMPLATE"] = False
+        toolchain.cache_variables["CMAKE_VERBOSE_MAKEFILE"] = True
+        # toolchain.cache_variables["CMAKE_CXX_COMPILER_LAUNCHER"] = "ccache"
+        # toolchain.cache_variables["CMAKE_C_COMPILER_LAUNCHER"] = "ccache"
+        toolchain.generate()
+
+    def validate(self):
+        # generic OpenVINO requirements
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, "11")
+        #     if self.options.enable_cpu:
+        #         # MLAS requires C++ 17
+        #         check_min_cppstd(self, "17")
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        compiler_version = Version(self.settings.compiler.version)
+        if minimum_version and compiler_version < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires {self.settings.compiler} ver. {minimum_version}, provided ver. {compiler_version}.",
+            )
+
+        if self.settings.os == "Emscripten":
+            raise ConanInvalidConfiguration(f"{self.ref} does not support Emscripten")
+
+        if self.settings.build_type == "Debug":
+            raise ConanInvalidConfiguration(f"{self.ref} does not support Debug build type")
+
+        # GPU does not support oneDNN in static build configuration, warn about it
+        if self.options.get_safe("enable_gpu") and not self.options.shared:
+            self.output.warning(f"{self.name} recipe builds GPU plugin without oneDNN (dGPU) support during static build.")
+
+    def build_validate(self):
+        if self._protobuf_required and self.options.shared and self.options["protobuf"].shared:
+            raise ConanInvalidConfiguration(f"{self.ref}:shared=True requires protobuf:shared=False for correct work.")
```

Comment:
@SSE4 @RubenRBS  I need to throw exception for cases when protobuf is shared. Current exception does not work, because shared protobuf is still used in this build https://c3i.jfrog.io/c3i/misc-v2/summary.html?json=https://c3i.jfrog.io/c3i/misc-v2/logs/pr/19164/77-linux-gcc/openvino/2023.1.0//summary.json 

How can I guarantee that only static one is used?